`Development`: Extend ngx-charts documentation section (#5341)
ls1intum/Artemis: docs/dev/guidelines/client.rst

       Depending on the chart type, there is more than one type of tooltip configurable.
       For more information visit https://swimlane.gitbook.io/ngx-charts/

    2. In order to manipulate the content of the data label (e.g. the text floating above a chart bar), the framework provides a ``[dataLabelFormatting]`` property in the
       HTML template that can be assigned to a method. For example:

       .. code-block:: html+ng2

          [dataLabelFormatting]="formatDataLabel"

       with

       .. code-block:: ts

>           formatDataLabel(averageScore: number): string {
>                   return averageScore + '%';
>               }

Reviewer: ```suggestion
          formatDataLabel(averageScore: number): string {
              return averageScore + '%';
          }
```
indentation

       appends a percentage sign to the data label.

       .. TIP::

>            The method is passed to the framework itself and executed there. This means on runtime, it does not have access to global variables of the component it is originally implemented in.

Reviewer: ```suggestion
           The method is passed to the framework itself and executed there. This means that at runtime it does not have access to global variables of the component it is originally implemented in.
```

       First, annotate the parent container with a reference (in the example ``#containerRef``).
       Then, when configuring the dimensions of the chart in ``[view]``, insert ``containerRef.offsetWidth`` instead
       of an specific value for the width.

    5. There are two ways to keep axis labels and axis ticks translation-sensitive if they contain natural language:

       * Axis labels are passed directly as property in the HTML template. Simply insert the translation string together with the translate pipe:

       .. code-block:: html+ng2

           [xAxisLabel]="'artemisApp.exam.charts.xAxisLabel' | artemisTranslate"
           [yAxisLabel]="'artemisApp.exam.charts.yAxisLabel' | artemisTranslate"

       * For some chart types, the framework derives the ticks of one axis from the name property of the passed data objects.
         So, these names have to be translated every time the user switches the language settings.
         In this case, inject the ``TranslateService`` to the underlying component and subscribe to the ``onLangChange`` event emitter:

       .. code-block:: ts

>            constructor(private translateService: TranslateService) {
>                 this.translateService.onLangChange.subscribe(() => {
>                     this.updateXAxisLabel(); // a method re-assigning the names of the objects to the translated string
>                 });
>            }

Reviewer: ```suggestion
           constructor(private translateService: TranslateService) {
               this.translateService.onLangChange.subscribe(() => {
                   this.updateXAxisLabel(); // a method re-assigning the names of the objects to the translated string
               });
           }
```
indentation
